### DISCUSSION TITLE
Is it possible to download the .idx files locally?

### DISCUSSION THREAD

**Discussion by tsilas3** [blaylockbk/Herbie#213] [Q&A]
I know I can use `.download()` with FastHerbie to download grib files themselves, but when creating a Herbie object for a given date/time/forecast hour, it still uses AWS to get the associated `.idx` files even if the grib files are available locally. Is there a way to download the `.idx` files in the same way? If I'm understanding correctly, this would mean I would be able to run Herbie code without an internet connection.

**Comment by blaylockbk** [comment]
Hi @tsilas3 

This is a great idea and one that would be beneficial to add to Herbie. I'll create a new issue (#215) to add this feature. It should be easy to make a flag `download_inventory=True` to always download the inventory files (thus enable an "offline" mode for Herbie). I will also have to add logic to tell Herbie to look for the inventory file locally before looking on the remote. (You'll see I thought about this some time ago, it just slipped through the cracks... https://github.com/blaylockbk/Herbie/blob/f48ba8c1ec34829195b3673fed00e26f57dd5bbf/herbie/core.py#L444-L445

---
<br><br>


Just some notes to myself...

It's not obvious how, but you can download your GRIB2 files locally, then use `_WGRIB2()` to make inventory files for every GRIB2 file in a path.

https://github.com/blaylockbk/Herbie/blob/f48ba8c1ec34829195b3673fed00e26f57dd5bbf/herbie/wgrib2.py#L46-L94

> Caveat: This requires the [wgrib2](https://anaconda.org/conda-forge/wgrib2) software be installed (not easily installed on Windows).

To use that you would do something like 

```python
# Download a file locally
from herbie import Herbie
H = Herbie("2023-7-17", model="hrrr", fxx=3)
H.download()

# Then create inventory files for all locally downloaded models
from herbie.wgrib2 import _WGRIB2
_WGRIB2().create_inventory_file("/home/blaylock/data/hrrr/20230717", suffix=".grib2")
```

This could be benefiicial for cases when there isn't an index file to download. Now, the problem is that Herbie doesn't currently know that it should look for the index files locally.
